### PR TITLE
Document bridging withdrawal process

### PR DESCRIPTION
**Pull request by timothymcmackin**
Not sure if talking about what the sequencer and the EVM nodes do is the right way to go about this, but it seemed better than just saying "etherlink does everything." If I keep it this way I should probably also give more detail about what Etherlink does in the deposit process.

Preview: https://docs-etherlink-git-bridging-withdrawal-trili-tech.vercel.app/building-on-etherlink/bridging

### CODE REVIEW

**Review by devon-n — APPROVED** [review]

File: docs/get-started/bridging.mdx [hunk 1]
```diff
@@ -26,12 +26,9 @@ The Etherlink indexer run by Nomadic Labs automatically executes these bridging
 
 ## Bridging your tokens
 
-Etherlink provides bridges for tez tokens:
+Etherlink provides a [Mainnet canonical bridge](https://bridge.etherlink.com/) for tez tokens.
```
Comment (lthms) [inline docs/get-started/bridging.mdx:29]: Maybe mention that the bridge is trustless and permissionless (no third party involved, everyone can use it without any restrictions)

File: docs/get-started/bridging.mdx [hunk 2]
```diff
@@ -71,6 +71,21 @@ The request includes the tez to bridge, the address of the Etherlink Smart Rollu
...
 
-This diagram is an overview of the process:
+This diagram is an overview of the deposit process:
+
+![Overview of the token bridging deposit process](/img/bridging-deposit.png)
+
+The withdrawal process follows these general steps:
+
+1. A user sends tez to the Etherlink Smart Rollup.
+The request includes their layer 1 address.
```
[the remark this answers is not in the document]
Comment (vch9) [reply docs/get-started/bridging.mdx:81]: Perhaps we could be more explicit (or technical?) and mention directly the precompiled contract that is responsible for emitting withdrawals

File: docs/get-started/bridging.mdx [hunk 3]
```diff
@@ -71,6 +71,21 @@ The request includes the tez to bridge, the address of the Etherlink Smart Rollu
 1. The bridge contract forwards the ticket to the stored Etherlink Smart Rollup address and clears its storage for the next transfer.
 1. The Etherlink Smart Rollup uses the information in the ticket to mint tokens on Etherlink and send them to the user's Etherlink address.
```
Comment (vch9) [inline docs/get-started/bridging.mdx:72]: We could add an additional line saying that a deposit on L1 creates a transfer on L2 from the zero address: https://explorer.etherlink.com/address/0x0000000000000000000000000000000000000000

File: docs/get-started/bridging.mdx (hunk 1 above)
Comment (timothymcmackin) [reply docs/get-started/bridging.mdx:29]: > This bridge is trustless and permissionless; anyone can use it without restrictions or the intervention of a third party.

**Comment by timothymcmackin** [comment]
@vch9 Thanks for clarifying how things work with the precompiled contract on the Etherlink side. Does this sound correct? 
https://github.com/etherlinkcom/docs/pull/101/commits/fb72e8c8dde9e6f3d6106887917aa689394343a2

**Comment by vch9** [comment]
> @vch9 Thanks for clarifying how things work with the precompiled contract on the Etherlink side. Does this sound correct? [fb72e8c](https://github.com/etherlinkcom/docs/commit/fb72e8c8dde9e6f3d6106887917aa689394343a2)

In comparison to the L1 contract worfklow, I find the "Etherlink" part a bit blurry. I don't know how technical and precise it should be though.

Here are a few changes I can propose:
```diff
diff --git a/docs/get-started/bridging.mdx b/docs/get-started/bridging.mdx
index 15e461a..a9d3b8f 100644
--- a/docs/get-started/bridging.mdx
+++ b/docs/get-started/bridging.mdx
@@ -70,8 +70,9 @@ The request includes the tez to bridge, the address of the Etherlink Smart Rollu
 1. The exchanger contract stores the tez and creates a [ticket](https://docs.tezos.com/smart-contracts/data-types/complex-data-types#tickets) that represents the receipt of the tokens.
 1. The exchanger contract sends the ticket to the bridge contract's `callback` entrypoint.
 1. The bridge contract forwards the ticket to the stored Etherlink Smart Rollup address and clears its storage for the next transfer.
-1. The Etherlink Smart Rollup transfers the ticket to a [precompiled contract](https://explorer.etherlink.com/address/0xff00000000000000000000000000000000000001) on Etherlink.
-1. The precompiled contract creates the tez and sends them to the user's Etherlink address.
+1. The Etherlink Smart Rollup receives the deposit request from the shared inbox.
+1. The Etherlink Smart rollup emits a corresponding transaction on Etherlink transferring the tez to the receiver address
+from the [zero address](https://explorer.etherlink.com/address/0x0000000000000000000000000000000000000000).
 
 This diagram is an overview of the deposit process:
 
@@ -79,12 +80,12 @@ This diagram is an overview of the deposit process:
 
 The withdrawal process follows these general steps:
 
-1. A user sends tez to a [precompiled contract](https://explorer.etherlink.com/address/0xff00000000000000000000000000000000000001) in the Etherlink Smart Rollup.
+1. A user sends tez to the [withdrawal precompiled contract](https://explorer.etherlink.com/address/0xff00000000000000000000000000000000000001) in the Etherlink Smart Rollup.
 The request includes their layer 1 address.
-1. The contract locks the tez and creates a [ticket](https://docs.tezos.com/smart-contracts/data-types/complex-data-types#tickets) that represents the receipt of the tokens.
-1. The contract puts the ticket in a call to the exchanger contract's `burn` entrypoint and puts this transaction in the Smart Rollup outbox on layer 1.
-The transaction exists in the outbox as part of Etherlink's commitment to its state.
-1. When the commitment that contains the transaction is cemented, anyone can run the transaction by running the Octez client `execute outbox message` command.
+1. The contract locks the tez.
+1. The contract emits an outbox message back to the exchanger contract's `burn` entrypoint and puts this transaction in the Smart Rollup outbox.
+The outbox message is part of Etherlink's commitment to its state.
+1. When the commitment that contains the transaction is cemented on L1, anyone can run the transaction by running the Octez client `execute outbox message` command.
 1. The exchanger contract receives the ticket, burns it, and sends the equivalent amount of tez to the user's layer 1 address.
 
 This diagram is an overview of the withdrawal process:
```

**Comment by vch9** [comment]
Actually the different nodes are documented in https://docs.etherlink.com/network/architecture, should we be precise about whether the transaction is in the evm-node or smart rollup node? It's relatively important w.r.t to tickets/inbox/outbox

File: docs/get-started/bridging.mdx [hunk 4]
```diff
@@ -41,36 +46,55 @@ The bridge uses these general steps:
...
 1. The exchanger contract stores the tez and creates a [ticket](https://docs.tezos.com/smart-contracts/data-types/complex-data-types#tickets) that represents the receipt of the tokens.
 1. The exchanger contract sends the ticket to the bridge contract's `callback` entrypoint.
-1. The bridge contract forwards the ticket to the stored Etherlink Smart Rollup address and clears its storage for the next transfer.
-1. The Etherlink Smart Rollup uses the information in the ticket to mint tokens on Etherlink and send them to the user's Etherlink address.
+1. The bridge contract forwards the ticket to the Smart Rollup inbox and clears its storage for the next transfer.
+1. Etherlink Smart Rollup nodes receive the deposit transaction from the Smart Rollup inbox.
+1. The Smart Rollup nodes put the deposit transaction in the delayed inbox.
+1. The sequencer requests the state of Etherlink from a Smart Rollup node and receives the delayed inbox.
+1. The sequencer creates a corresponding transaction on Etherlink for a precompiled contract to transfer the tez from the [zero address](https://explorer.etherlink.com/address/0x0000000000000000000000000000000000000000) to the user's address.
```
[the remark this answers is not in the document]
Comment (timothymcmackin) [reply docs/get-started/bridging.mdx]: I'm sorry, I don't follow. The precompiled contract in the deposit process diagram is what sends the  tez to the user's EVM account. Is that  the same precompiled contract in the withdraw process?

File: docs/get-started/bridging.mdx [hunk 5]
```diff
@@ -41,36 +46,55 @@ The bridge uses these general steps:
...
-  - This contract is deployed to Ghostnet at [`KT1VEjeQfDBSfpDH5WeBM5LukHPGM2htYEh3`](https://ghostnet.tzkt.io/KT1VEjeQfDBSfpDH5WeBM5LukHPGM2htYEh3/).
+  - This contract is deployed to testnet at [`KT1VEjeQfDBSfpDH5WeBM5LukHPGM2htYEh3`](https://ghostnet.tzkt.io/KT1VEjeQfDBSfpDH5WeBM5LukHPGM2htYEh3/).
   - This contract is deployed to Mainnet at [`KT1CeFqjJRJPNVvhvznQrWfHad2jCiDZ6Lyj`](https://tzkt.io/KT1CeFqjJRJPNVvhvznQrWfHad2jCiDZ6Lyj/).
 
-1. A user sends a request to the bridge contract's `deposit` entrypoint.
+### Deposit process
+
+The deposit process (moving tez from layer 1 to Etherlink) follows these general steps:
+
+1. A Tezos user sends a request to the layer 1 bridge contract's `deposit` entrypoint.
```
[the remark this answers is not in the document]
Comment (timothymcmackin) [reply docs/get-started/bridging.mdx:69]: How about this:

> This bridge contract is not a fundamental part of the bridge; it is a helper contract that avoids limitations around tickets by forwarding them to the Etherlink Smart Rollup on behalf of user accounts.

File: docs/get-started/bridging.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (vch9) [reply docs/get-started/bridging.mdx]: It's not a precompiled contract in the deposit. A precompiled contract is a special kind of contract bundled with the EVM at fixed address that developers might use in their contracts. Ethereum defines an exhaustive list of precompiled contracts, we added an additional one for the withdrawal at the address `0xff00..01`.

For the deposit there's no precompiled contract, it's the core logic of the kernel that crafts a transfer from the zero address to the user's EVM account.

[the remark this answers is not in the document]
Comment (timothymcmackin) [reply docs/get-started/bridging.mdx]: Updated the diagram to show that it's not a precompiled contract but the kernel that creates the transfer from the zero address to the user's EVM account.

File: docs/building-on-etherlink/bridging.mdx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+1. It sends the tez in a transaction to the exchanger contract's `mint` entrypoint.
+1. The exchanger contract stores the tez and creates a [ticket](https://docs.tezos.com/smart-contracts/data-types/complex-data-types#tickets) that represents the receipt of the tokens.
+1. The exchanger contract sends the ticket to the bridge contract's `callback` entrypoint.
+1. The bridge contract forwards the ticket to the Smart Rollup inbox and clears its storage for the next transfer.
+1. Etherlink Smart Rollup nodes receive the deposit transaction from the Smart Rollup inbox.
+1. The Smart Rollup nodes put the deposit transaction in the delayed inbox.
+1. The sequencer requests the state of Etherlink from a Smart Rollup node and receives the delayed inbox.
+1. The sequencer creates a corresponding transaction on Etherlink to transfer tez from the [zero address](https://explorer.etherlink.com/address/0x0000000000000000000000000000000000000000) to the user's address.
+1. The sequencer adds this transaction to a blueprint as in the usual transaction lifecycle described in [Architecture](../network/architecture).
+1. The precompiled contract runs the transaction to transfer the tez.
```
Comment (vch9) [inline docs/building-on-etherlink/bridging.mdx:83]: This line should be removed

Comment (timothymcmackin) [reply docs/building-on-etherlink/bridging.mdx]: Removed, thx.

**Pull request merged**